pkg/cmd/server/start/start_node.go

```diff
@@ -225,17 +226,17 @@ func (o NodeOptions) CreateNodeConfig() error {
 	return nil
 }
 
-func RunSDNController(config configapi.NodeConfig) {
+func RunSDNController(config configapi.NodeConfig, nodeConfig *kubernetes.NodeConfig) {
 	if config.NetworkPluginName == osdn.NetworkPluginName() {
-		kclient, _, err := configapi.GetKubeClient(config.MasterKubeConfig)
-		if err != nil {
-			glog.Fatal("Failed to get kube client for SDN")
-		}
 		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
 		if err != nil {
 			glog.Fatal("Failed to get kube client for SDN")
 		}
-		osdn.Node(*oclient, *kclient, config.NodeName, "")
+		ch := make(chan struct{})
+		nodeConfig.KubeletConfig.StartUpdates = ch
+		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
+		// no plugin necessary
+		nodeConfig.KubeletConfig.NetworkPluginName = ""
```

Comment:
@rajatchopra rajat, https://github.com/openshift/origin/pull/2984/files#diff-ef4c2b4de3a6a004d532acc3fccf5779R210 set the plugin in the nodeConfig... @smarterclayton is there a better way to plumb it through?